### PR TITLE
Updates go fluent path api

### PR DESCRIPTION
**Pull request by rkodev**
## Overview
Resolves https://github.com/microsoftgraph/msgraph-sdk-go/issues/541

Fixes Go samples fluent api.

### Demo

```go
 # https://learn.microsoft.com/en-us/graph/api/worksheet-post-charts?view=graph-rest-1.0&tabs=http
 // current syntax 
result, err := graphClient.Drives().ByDriveId("drive-id").Items().ByItemId("driveItem-id").Workbook().Worksheets().ByWorksheetId("workbookWorksheet-id").Charts().Post(context.Background(), requestBody, nil)


// expected

result, err := graphClient.Drives().ByDriveId("drive-id").Items().ByDriveItemId("driveItem-id").Workbook().Worksheets().ByWorkbookWorksheetId("workbookWorksheet-id").Charts().Post(context.Background(), requestBody, nil)rkbook().Worksheets().ByWorksheetId("workbookWorksheet-id").Charts().Post(context.Background(), requestBody, nil)

```

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* Added unit tests

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 1]
```diff
@@ -565,7 +565,10 @@ private static string GetFluentApiPath(IEnumerable<OpenApiUrlTreeNode> nodes)
             var elements = nodes.Select(static (x, i) =>
             {
                 if (x.Segment.IsCollectionIndex())
-                    return $"ByTypeId{x.Segment.Replace("{", "(\"").Replace("}", "\")")}.";
+                {
+                    var pathName = x.Segment?.Replace("{","").Replace("}","").Split('-').Where(static s => !string.IsNullOrEmpty(s)).Select(static s => s.ToFirstCharacterUpperCase()).Aggregate(static (a, b) => $"By{a}{b}");
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:569]: https://learn.microsoft.com/en-us/dotnet/api/system.string.replace?view=net-7.0#system-string-replace(system-string-system-string-system-stringcomparison)

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection/StringExtensions/StringExtensions.cs [hunk 2]
```diff
@@ -31,6 +31,19 @@ public static class StringExtensions
...
 
+        public static string ReplaceMultiple(this string stringValue, string haystack, params string []needles)
+        {
+            // loop over stringValue check if needle exists before replacing it with haystack value
+            if (string.IsNullOrEmpty(stringValue)) return stringValue;
+            if (needles == null || needles.Length == 0) return stringValue;
+            foreach (var needle in needles)
+            {
+                if (string.IsNullOrEmpty(needle) || !stringValue.Contains(needle, System.StringComparison.InvariantCulture)) continue;
+                stringValue = stringValue.Replace(needle, haystack);
```
Comment (baywet) [inline CodeSnippetsReflection/StringExtensions/StringExtensions.cs:42]: string comparison options for all the replace

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 3]
```diff
@@ -565,7 +565,10 @@ private static string GetFluentApiPath(IEnumerable<OpenApiUrlTreeNode> nodes)
             var elements = nodes.Select(static (x, i) =>
             {
                 if (x.Segment.IsCollectionIndex())
-                    return $"ByTypeId{x.Segment.Replace("{", "(\"").Replace("}", "\")")}.";
+                {
+                    var pathName = string.IsNullOrEmpty(x.Segment) ? x.Segment : x.Segment.ReplaceMultiple("", "{", "}").Split('-').Where(static s => !string.IsNullOrEmpty(s)).Select(static s => s.ToFirstCharacterUpperCase()).Aggregate(static (a, b) => $"By{a}{b}");
+                    return $"{pathName ?? "ByTypeId"}{x.Segment.Replace("{", "(\"").Replace("}", "\")")}.";
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:570]: comparison options

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs (hunk 3 above)
Comment (rkodev) [reply CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs]: done

**Review by baywet — APPROVED** [review]
thanks for making the changes!

**Pull request merged**